### PR TITLE
Won/home screen flow

### PR DESCRIPTION
**Pull request by wkim10**
#11 

Added pages + backend for "Join Class" as shown in the Figma section 1.

Testing:
- Go to /login and sign in, which takes you to /course
- Click "Join class" which takes you to /course/join
- Try inputting different types of class codes into the text field (test out edge cases)
- Can add new classes into Firebase or test using existing classes (with new "code" field)

Some images:
Screen Shot 2024-08-16 at 10 48 41 PM
Screen Shot 2024-08-16 at 10 46 26 PM
Screen Shot 2024-08-16 at 10 46 52 PM
Screen Shot 2024-08-16 at 10 47 09 PM

### CODE REVIEW

File: src/app/private/course/join/page.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+              Join
+            </Typography>
+          </Button>
+        }
+      />
+      <div className="mx-4 mt-12">
+        <div>
+          Ask your TA or professor for the class code, then enter it here.
+        </div>
+        <TextField
```
Comment (nickbar01234) [inline src/app/private/course/join/page.tsx:159]: Probably should disable auto complete / suggestions

File: src/app/private/course/join/page.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        }
+        title="Join class"
+        rightIcon={
+          <Button
+            className="py-2.5 px-6 bg-[#38608F] rounded-full mr-2 normal-case"
+            onClick={joinClicked}
+          >
+            <Typography variant="subtitle2" color="#FFFFFF" fontWeight={600}>
+              Join
+            </Typography>
```
Comment (johnny-t06) [inline src/app/private/course/join/page.tsx:151]: Can we make the button disabled until the entered code length is valid

**Review by johnny-t06 — APPROVED** [review]
LGTM, tested all the pages and the code join cases. Looks great, thanks Won!
